### ISSUE TITLE
eom window covers the screen for certain resolutions 

### ISSUE THREAD

**Issue by Botspot** [mate-desktop/eom#349]
#### Expected behaviour
For images that are too large to display at 100% zoom level on this display, eom should display the image at a smaller size and scale the window appropriately.

#### Actual behaviour
For certain image resolutions, eom fails to scale down the image and instead covers the entire screen with a window larger than the physical screen size.

#### Steps to reproduce the behaviour
Open an image with the following dimensional characteristics:

- Height of image is greater than the screen height
- Height of image is an even number of pixels.
- Width of image is an odd number of pixels.

Here is an image with those characteristics: [2263x2262.jpg](https://github.com/RPi-Distro/repo/assets/54716352/38365ebe-be45-4793-a7bd-eff2118a6332)

On my 1080p screen, opening the image looks like this:
![300643870-70889401-a7e7-44d8-b480-e05e7b61ba74](https://github.com/RPi-Distro/repo/assets/54716352/93ebc0a5-dc2d-42b1-a3db-2c81674a8187)
You can see that eom's interface and the window border is beyond the top and bottom edges of the screen. The only way to close it was with Alt+F4.

This may seem like an extreme edge case, but I have a number of manually cropped images with an odd X-dimension and an even Y-dimension, thanks to Shotwell's not-quite-perfect job at cropping pictures to a square. 

#### MATE general version
Mate is not installed on this distro.

#### Package version
eom 1.26.0-2
`Maintainer: Debian+Ubuntu MATE Packaging Team <debian-mate@lists.debian.org>`

#### Linux Distribution
Raspberry Pi OS Bookworm 64-bit (wayfire window manager)

#### Link to bugreport of your Distribution (requirement)
https://github.com/RPi-Distro/repo/issues/366

**Comment by Botspot** [comment]
This should be really easy to reproduce and an easy fix. Someone please respond.
@raveit65
`Pinging @whs @mbkma @rbuj next if @raveit65 does not respond`

**Comment by lukefromdc** [comment]
I can confirm that larger images than the screen height overflow and that this is not limited to the Pi or even to x11. I just opened a 4048x3048 image on a 4K monitor, and the window overflowed both top and bottom. Note that in compiz you can grab the window from the "expo" view and in wayfire (wayland) it's even simpler: for that case <super> right mouse button will drag any window from anywhere on it.

Obviously we need to check the monitor dimensions before opening an image. GdkScreen should NOT be used, as two monitors set up one partially above and left or right of another will yield the smallest "screen" containing both monitors

**Comment by Botspot** [comment]
Can confirm the overflow on my 1080p screen with a 4048x3048 image. It appears that my odd/even hypothesis stated above was incorrect. Though it is interesting to note that EOM does display at a more rational size after resizing the image to either 4048x4048 or 3048x3048.

**Comment by lukefromdc** [comment]
If ANY image larger than the monitor stays contained within the monitor, that implies we have monitor size checking code that works some of the time, but not all of the time. I haven't done much of the work on EOM so I don't know offhand where it would be. Fixing this will still need attention from others on the team

**Comment by xBZZZZ** [comment]
>we have monitor size checking code that works some of the time, but not all of the time

there is but it is weird (why <a href="https://github.com/mate-desktop/eom/blob/36f011e8138655675e43f29c4e21b79dfc7c8e4b/src/eom-window.c#L1266"><code>if (img_width > img_height) {</code></a>?) and wrong (window doesn't fit on screen)

I have rewritten it: https://github.com/mate-desktop/eom/pull/380

does my rewrite work on wayfire?

**Comment by Botspot** [comment]
Yes that does look like a genuine logic bug. Can you look at it @whs @mbkma @rbuj?

**Comment by Botspot** [comment]
Looks like the commit that added this was https://github.com/mate-desktop/eom/commit/a8d28a6ce7e0c56dacba5d527d9134573a008902 by @incial

**Comment by lukefromdc** [comment]
> Looks like the commit that added this was [a8d28a6](https://github.com/mate-desktop/eom/commit/a8d28a6ce7e0c56dacba5d527d9134573a008902) by [@incial](https://github.com/incial)

That looks like the initial upload or very close to it
